examples/text-classification/run_glue.py

```diff
@@ -133,12 +132,51 @@ class ModelArguments:
     )
 
 
+# Should be moved to transformers/src/transformers/file_utils.py
+def is_run_on_sagemaker():
+    if "SM_OUTPUT_DATA_DIR" in os.environ and "SM_MODEL_DIR" in os.environ:
+        return True
+    else:
+        return False
+
+
+# Should be moved to transformers/src/transformers/file_utils.py or transformers/src/transformers/hf_argparser.py
+def parse_sagemaker_args(argv):
+    ## add output_dir
+    argv.extend(["--output_dir", os.environ["SM_OUTPUT_DATA_DIR"]])
+    # if datafiles add them as args
+    for key, value in os.environ.items():
+        # get all passed in s3 data_paths
+        if key.startswith("SM_CHANNEL_"):
+            # extract channel
+            key_type = key.split("_")[-1].lower()
+            # check if sm_channel is a real file with *.csv or *.json or if it is a path
+            if value.endswith(".csv") or value.endswith(".json"):
+                # if true add args --{channel}_file with value
+                argv.extend([f"--{channel}_file", value])
+            else:  # if train file is passed hyperparameter and channel is a directory
+                # get index of {channel}_file and add +1 to get the file path
+                index_of_file_from_key_type = argv.index(f"--{key_type}_file") + 1
+                # create new path from channel and file name
+                new_path = os.path.join(value, argv[index_of_file_from_key_type])
+                # overwrite existing argument for {channel}_file
+                argv[index_of_file_from_key_type] = new_path
+    ## remove True for Argsparser to work
+    argv = [args for args in argv if args != "True"]
+    return argv
+
+
 def main():
     # See all possible arguments in src/transformers/training_args.py
     # or by passing the --help flag to this script.
     # We now keep distinct sets of args, for a cleaner separation of concerns.
-
+    # ##### $custom ####
+    if is_run_on_sagemaker():
+        sys.argv = parse_sagemaker_args(sys.argv)
+        print(sys.argv)
```

Comment:
We should use the logger instead of just printing. Also, those are going to be logged anyway a little bit later, so is this just temporary for debugging purposes?

Reply:
Sorry for that, yeah the print was only for developing and debugging.